src/setup.c

```diff
@@ -81,6 +85,60 @@ static void setup_gpio(void)
 	/* Speaker */
 	gpio_mode_setup(GPIOB, GPIO_MODE_AF, GPIO_PUPD_NONE, GPIO9);
 	gpio_set_af(GPIOB, GPIO_AF3, GPIO9);
+
+	/* Motor driver */
+	gpio_mode_setup(GPIOC, GPIO_MODE_AF, GPIO_PUPD_NONE,
+			GPIO6 | GPIO7 | GPIO8 | GPIO9);
+	gpio_set_af(GPIOC, GPIO_AF3, GPIO6 | GPIO7 | GPIO8 | GPIO9);
+}
+
+/**
+ * @brief Setup PWM for the motor drivers.
+ *
+ * TIM8 is used to generate both PWM signals (left and right motor):
+ *
+ * - Edge-aligned, up-counting timer.
+ * - Prescale to increment timer counter at 24 MHz.
+ * - Set PWM frequency to 24 kHz.
+ * - Configure channels 1, 2, 3 and 4 as output GPIOs.
+ * - Set output compare mode to PWM1 (output is active when the counter is
+ *   less than the compare register contents and inactive otherwise.
+ * - Reset output compare value (set it to 0).
+ * - Enable channels 1, 2, 3 and 4 outputs.
+ * - Enable outputs in the break subsystem.
+ * - Enable timer counter.
+ *
+ * @see Reference manual (RM0090) "Advanced-control timers (TIM1 and TIM8)"
+ * and in particular the "PWM mode" section.
+ */
+static void setup_motor_driver(void)
+{
+	timer_set_mode(TIM8, TIM_CR1_CKD_CK_INT, TIM_CR1_CMS_EDGE,
+		       TIM_CR1_DIR_UP);
+
+	timer_set_prescaler(
```

Comment:
I'm not sure if the general changes here, for example the timer_set_prescaler value should be also moved to bulebule project.. (pull request, issue?)

Reply:
Which changes do you mean?

Reply:
Forget this, I was thinking on the prescaler formula but it is also on Bulebule project, I mixed functions.